### PR TITLE
docs - update docs contribution instructions

### PR DESCRIPTION
**Pull request by briantist**
##### SUMMARY

Update the Contribution Guide for docs submissions.

##### ISSUE TYPE

- Docs Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: docs/docsite/rst/contributor_guide.rst [hunk 1]
```diff
@@ -30,23 +30,25 @@ Contributing documentation
...
+Markdown files (those with the extension ``.md``) can be found in several directories within the repository. These files are primarily aimed at developers and those browsing the repository, to explain or give context to the other files nearby.
+
+The main exception to the above is the ``README.md`` in the repository root. This file is more important because it provides introductory information and links for anyone browsing the repository, both on GitHub and on the collection's `Ansible Galaxy page <https://galaxy.ansible.com/community/hashi_vault>`_.
 
-This type of documentation is tested with ``ansible-test sanity`` and full instructions are available on the :ref:`testing module documentation <testing_module_documentation>` page.
+Markdown files can be previewed natively in GutHub, so they are easy to validate by reviewers, and there are no specific tests that need to run against them.
```
Comment (bmillemathias) [inline docs/docsite/rst/contributor_guide.rst:40]: typo "GutHub".

Comment (briantist) [reply docs/docsite/rst/contributor_guide.rst:40]: 🤦‍♂️I have been doing this a LOT lately, thank you.

**Review by samccann — APPROVED** [review]
LGTM

**Comment by briantist** [comment]
thank you for reviewing @samccann and @bmillemathias !

**Pull request merged**